src/core/runner.ts

```diff
@@ -366,6 +368,7 @@ export default class Runner {
           result.error = stepResult.error;
         }
       }
+      result.steps = stepResults;
```

Comment:
I would remove it and test using the custom reporter hooks since its mainly used to make the testing easier.

Reply:
this will also enhance creating custom reporter detail reporting.

Reply:
Sure, makes sense.